extras/rya.streams/query-manager/src/main/java/org/apache/rya/streams/querymanager/QueryManager.java

```diff
@@ -0,0 +1,263 @@
+                        LOG.debug("delete query event.");
+                        if (newQueryState.isPresent()) {
+                            stopQuery(newQueryState.get().getQueryId());
+                            LOG.info("Deleted query: " + newQueryState.get().toString());
+                        } else {
+                            LOG.debug("Delete requested a query that does not exist yet.");
+                        }
+                        break;
+                    case UPDATE:
+                        LOG.debug("update query event.");
+                        if (!newQueryState.isPresent()) {
+                            LOG.debug("Query: " + entry.getQueryId() + " does not exist yet, cannot perform update.");
+                        } else {
+                            final StreamsQuery updateQuery = newQueryState.get();
+                            // if the query is currently inactive, and is updated to
+                            // be active, turn on.
+                            if (!updateQuery.isActive() && entry.getIsActive().or(false)) {
+                                final StreamsQuery newQuery = new StreamsQuery(entry.getQueryId(), entry.getSparql().get(),
+                                        entry.getIsActive().or(false));
+                                runQuery(ryaInstanceName, newQuery);
+                                LOG.info("Starting query: " + newQuery.toString());
+                            } else if (!updateQuery.isActive() && !entry.getIsActive().or(true)) {
+                                // if the query is running and the update turns it
+                                // off, turn off.
+                                stopQuery(updateQuery.getQueryId());
+                                LOG.info("Stopping query: " + updateQuery.toString());
+                            } else {
+                                LOG.debug("The query is either already running and "
+                                        + "updated to turn on, or is already stopped and is updated to stop running.");
+                            }
+                        }
+                        break;
+                }
+            } finally {
+                lock.unlock();
+            }
+        }
+    }
+
+    /**
+     * Listener used by the {@link QueryManager} to be notified when
+     * {@link QueryChangeLog}s are created or deleted.
+     */
+    private class QueryManagerSourceListener implements SourceListener {
+        @Override
+        public void notifyCreate(final String ryaInstanceName, final QueryChangeLog log) {
+            lock.lock();
+            try {
+                LOG.info("Discovered new Query Change Log for Rya Instance " + ryaInstanceName + " within source " + log.toString());
+                final QueryRepository repo = new InMemoryQueryRepository(log, scheduler);
+                repo.startAndWait();
+                repo.subscribe(new QueryManagerQueryChange(ryaInstanceName));
```

Comment:
Doesn't this return a list StreamsQuery objects that need to be handled?

Reply:
yes....